src/data/nat/prime.lean

```diff
@@ -492,6 +492,26 @@ theorem prime.not_dvd_mul {p m n : ℕ} (pp : prime p)
   (Hm : ¬ p ∣ m) (Hn : ¬ p ∣ n) : ¬ p ∣ m * n :=
 mt pp.dvd_mul.1 $ by simp [Hm, Hn]
 
+/-- Prime `p` divides the product of `L : list ℕ` iff it divides some `a ∈ L` -/
+lemma prime.dvd_prod_iff {p : ℕ} {L : list ℕ} (pp : prime p) :
```

Comment:
I guess these lemmas also hold for prime elements in more general settings (comm_rings? comm_monoid_with_zero?) I think it would make sense to reuse those proofs. Would that work for your application?

Reply:
Ah, I should have thought of this after @ericrbg asked for the corresponding generalisation on #10675.  I've added the proof to `algebra/associated`, but I'm not sure if that's the right place for it.

Reply:
Hmm, on the basis of this thread (https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/mathlib.20workflow) I suspect that's maybe not the best place to put it.  I've moved it to a new file for now.

Reply:
Hmm (x2), now the original lemma (for `list ℕ`) seems to need `import ring_theory.int.basic` in order to use `nat.prime_iff`.  I don't want to add an extra import to such a basic file.

Until I can figure out what should go where I'm just moving everything to the temporary file.

Reply:
Ah, I should have thought of this after @ericrbg asked for the corresponding generalisation on #10675. I've added the proof for `comm_monoid_with_zero` to a temp file for now, because I'm not sure where it best belongs. `algebra/associated` perhaps?

I'm not sure about how to use the more general lemma to prove the version for `list ℕ`, since `data/nat/prime` can't see `nat.prime_iff`.  I don't want to add an extra import (of `ring_theory.int.basic`) to such a basic file just for this one lemma.

Reply:
I do still wonder whether something like "preprime" is a good idea. This, and many similar lemmas, hold for 0 too; although, to be fair, they go further to equality, so it may still be worth special casing them. I'm not sure, it's a random idea for the futurel

Reply:
Would it be possible to incorporate that into the unification proposed in this thread (https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/nat.2Eprime)?

Reply:
I've moved the more general proof for `comm_monoid_with_zero` to `algebra/associated` and deleted the temp file. I'm still not sure how to use this to prove the version for `ℕ`, but perhaps that can be done as a separate PR after the merging of `nat.prime` and `prime` as discussed in the above Zulip thread?

Reply:
I think it's worth keeping the separate proofs (maybe making a note that it's duplicated above both for now as we're about to refactor) and we can deal with the aftermath later.

Reply:
I've added a TODO comment to that effect at the site of the lemma in `data/nat/prime`.